src/caffe/proto/caffe.proto

```diff
@@ -745,6 +746,14 @@ message InnerProductParameter {
   optional int32 axis = 5 [default = 1];
 }
 
+message KeyValueDataParameter {
+  // CSV file with ';' delimiter that specifies which keys to read from the 
```

Comment:
Could the delimiter not also be part of the protocol definition, with ";" the default?